File: apps/mobile/test/backend/antigravity_catalog_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:vicoa/backend/agent_catalog.dart';
import 'package:vicoa/components/agent_type_icon/agent_type_icon_widget.dart';
import 'package:vicoa/custom_code/actions/api_resume_session.dart';

void main() {
  group('antigravity catalog entry', () {
    test('is in the fallback catalog with the four launch-flag modes', () {
      final catalog = agentCatalogFallback();
      final agent = catalog.agentById('antigravity');
      expect(agent?.label, 'Antigravity');
      // default -> request-review, acceptEdits -> --mode accept-edits,
      // plan -> --mode plan, bypassPermissions -> --dangerously-skip-permissions.
      expect(
        agent!.permissionModes.map((m) => m.id).toList(),
        <String>['default', 'acceptEdits', 'plan', 'bypassPermissions'],
      );
      expect(SessionConfig.defaultsFor(catalog, 'antigravity').permissionMode, 'default');
    });

    test('has no thinking picker: effort is baked into every agy model id', () {
      final catalog = agentCatalogFallback();
      final agent = catalog.agentById('antigravity')!;
      expect(agent.thinkingEfforts, isEmpty);
      expect(agent.models!.any((m) => m.id == 'gemini-3.8-flash-high'), isTrue);
      expect(SessionConfig.defaultsFor(catalog, 'antigravity').model, 'default');
    });

    test('toSpawnMetadata passes model + permission mode, never the sentinel', () {
      expect(
        SessionConfig(agent: 'antigravity', model: 'claude-sonnet-4-6', permissionMode: 'acceptEdits')
            .toSpawnMetadata(),
        {'model': 'claude-sonnet-4-6', 'permission_mode': 'acceptEdits'},
      );
      expect(SessionConfig(agent: 'antigravity', model: 'default').toSpawnMetadata(), {});
    });

    test('has a brand mark', () {
      expect(agentTypeHasLogo('Antigravity'), isTrue);
    });
  });

  group('resuming an antigravity session', () {
    test('resolves the slug from the display name', () {
      expect(resumeAgentSlug('Antigravity'), 'antigravity');
    });

    test('carries the agy conversation id into the relaunch', () {
      expect(
        resumeAgentSessionHandle({'antigravity_conversation_id': 'c9065f92-851f'}),
        'c9065f92-851f',
      );
    });
  });
}

File: apps/web/lib/antigravity-catalog.test.ts
import { describe, expect, test } from 'vitest';
import {
  AGENT_CATALOG_FALLBACK,
  agentById,
  defaultsFor,
  toSpawnMetadata,
} from './agent-catalog';
import { getAgentLogoSrc } from '@/components/dashboard/agent-type-icon';
import { agentSessionHandle, resumeAgentSlug } from './session-resume';

describe('antigravity catalog entry', () => {
  test('is in the fallback catalog with the four launch-flag permission modes', () => {
    const agent = agentById(AGENT_CATALOG_FALLBACK, 'antigravity');
    expect(agent?.label).toBe('Antigravity');
    // default -> request-review, acceptEdits -> --mode accept-edits,
    // plan -> --mode plan, bypassPermissions -> --dangerously-skip-permissions.
    expect(agent?.permission_modes?.map((m) => m.id)).toEqual([
      'default',
      'acceptEdits',
      'plan',
      'bypassPermissions',
    ]);
    expect(defaultsFor(AGENT_CATALOG_FALLBACK, 'antigravity').permission_mode).toBe('default');
  });

  test('has no thinking picker: effort is baked into every agy model id', () => {
    // `--effort` conflicts with a suffixed Gemini id and is rejected outright
    // by the Claude ids, so the model picker IS the effort picker.
    const agent = agentById(AGENT_CATALOG_FALLBACK, 'antigravity');
    expect(agent?.thinking_efforts).toBeUndefined();
    expect(agent?.models?.some((m) => m.id === 'gemini-3.8-flash-high')).toBe(true);
    expect(agent?.models?.some((m) => m.id === 'gemini-3.8-flash-low')).toBe(true);
  });

  test('starts on the "keep agy\'s own model" sentinel and does not send it', () => {
    expect(defaultsFor(AGENT_CATALOG_FALLBACK, 'antigravity').model).toBe('default');
    expect(toSpawnMetadata({ agent: 'antigravity', model: 'default', permission_mode: 'default' })).toEqual({
      permission_mode: 'default',
    });
  });

  test('sends an explicit model and permission mode', () => {
    expect(
      toSpawnMetadata({
        agent: 'antigravity',
        model: 'claude-sonnet-4-6',
        permission_mode: 'acceptEdits',
      })
    ).toEqual({ model: 'claude-sonnet-4-6', permission_mode: 'acceptEdits' });
  });

  test('does not offer steer', () => {
    expect(agentById(AGENT_CATALOG_FALLBACK, 'antigravity')?.supports_steer).toBeFalsy();
  });

  test('has a brand mark', () => {
    expect(getAgentLogoSrc('Antigravity')?.src).toBe('/images/integrations/antigravity.svg');
    expect(getAgentLogoSrc('antigravity')?.alt).toBe('Antigravity');
  });
});

describe('resuming an antigravity session', () => {
  test('resolves the slug from the display name when session_config is absent', () => {
    expect(
      resumeAgentSlug({ id: 'x', status: 'COMPLETED', agent_type_name: 'Antigravity' })
    ).toBe('antigravity');
  });

  test('carries the agy conversation id into the relaunch', () => {
    expect(
      agentSessionHandle({
        id: 'x',
        status: 'COMPLETED',
        instance_metadata: { antigravity_conversation_id: 'c9065f92-851f-447c' },
      })
    ).toBe('c9065f92-851f-447c');
  });
});
